Authenticators API: Deprecate docs from VuePress (#4924)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/oie-upgrade-mfa-enroll-policy/main/index.md

## Overview

> Okta Identity Engine sets apart factors and authenticators to align with industry standards:

Reviewer: ```suggestion
Identity Engine sets apart factors and authenticators to align with industry standards:
```

Author: @susanharper-okta This is the first mention of Identity Engine on the page.

- Identity Engine uses authenticators in its authenticator enrollment policy settings.

> - Okta Classic Engine uses factors in its Multifactor (MFA) enrollment policy settings.

Reviewer: ```suggestion
- Classic Engine uses factors in its multifactor (MFA) enrollment policy settings.
```

In Identity Engine, the MFA Enrollment Policy name has changed to [authenticator enrollment policy](/docs/reference/api/policy/#authenticator-enrollment-policy). Classic Engine still refers to the same policy as the [Multifactor (MFA) Enrollment Policy](/docs/reference/api/policy/#multifactor-mfa-enrollment-policy). In the API, the policy type of `MFA_ENROLL` remains unchanged, however, the `settings` data contains authenticator or factors, depending on the configuration.

> After your upgrade your org to Identity Engine, consider the following:

Reviewer: ```suggestion
After you upgrade your org to Identity Engine, consider the following:
```

- New authenticator enrollment policies that you create in the Admin Console are configured with authenticators.

- Existing authenticator enrollment policies that you created before upgrading to Identity Engine are still configured with factors.

- Existing authenticator enrollment polices that you save in the Admin Console have their factors converted to authenticators. The conversion is seamless to admin users that manage policies in the Admin Console.

> > **Note:** Whether you edit the authenticator enrollment policy or not, when you click **Update Policy** in the Admin Console, the factor-to-authenticator conversion occurs.

Reviewer: ```suggestion
> **Note:** Whether you edit the authenticator enrollment policy, when you click **Update Policy** in the Admin Console, the factor-to-authenticator conversion occurs.
```

> Existing code that manages and uses authenticator enrollment policies through the [Policy API](/docs/reference/api/policy/) requires some development work. After you upgrade your org, handle the factor to authenticator conversion schema changes. This guide provides key API considerations to upgrade your app for multifactor enrollment flows in Identity Engine.

Reviewer: ```suggestion
Existing code that manages and uses authenticator enrollment policies through the [Policy API](/docs/reference/api/policy/) requires some development work. After you upgrade your org, address the factor-to-authenticator conversion schema changes. This guide provides key API considerations to upgrade your app for multifactor enrollment flows in Identity Engine.
```

> **Note:** Another indicator of an authenticator-based enrollment policy is when `type=AUTHENTICATORS` in the [settings schema](/docs/reference/api/policy/#policy-settings-data-2). The `type` property could be absent from the response of factor-based authenticator enrollment policies.

> If the returned policy uses `authenticators`, you need to loop through the list of [Policy Authenticator objects](/docs/reference/api/policy/#policy-authenticator-object) and use the `key` property to identify the authenticator. See the [Authenticators API](https://developer.okta.com/docs/api/openapi/okta-management/management/tag/Authenticator/) for more details on the available authenticators in your org.

Reviewer: ```suggestion
If the returned policy uses `authenticators`, loop through the list of [Policy Authenticator objects](/docs/reference/api/policy/#policy-authenticator-object) and use the `key` property to identify the authenticator. See the [Authenticators API](https://developer.okta.com/docs/api/openapi/okta-management/management/tag/Authenticator/) for more details on the available authenticators in your org.
```

If you need to convert an existing factor-based authenticator enrollment policy to use authenticators, then update the policy with authenticators in the `settings` parameter. Use the [Policy Factors Configuration object and Authenticator keys mapping](#policy-factors-configuration-object-and-authenticator-keys-mapping) table to map Policy Factors Configuration objects to authenticator keys. See the [settings conversion example](#authenticator-enrollment-policy-settings-conversion-example).

You can revert to the factor-based authenticator enrollment policy model. Modify an existing authenticator-based enrollment policy to a factored one. To convert the authenticator keys to Policy Factor Configuration objects, see the following:

* [Policy Factors Configuration objects](/docs/reference/api/policy/#policy-factors-configuration-object)
* [Policy Factors Configuration object and Authenticator keys mapping](#policy-factors-configuration-object-and-authenticator-keys-mapping)

> > **Note:** Your app might be integrated with systems like Terraform that can’t be updated to parse the authenticators-based enrollment policy. If so, you need to revert your policy to use factors instead of authenticators.

Reviewer: ```suggestion
> **Note:** Your app might be integrated with systems like Terraform that you can't update to parse the authenticators-based enrollment policy. If so, revert your policy to use factors instead of authenticators.
```
